plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -243,28 +243,6 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             active_producers.insert(p.producer_name);
          }
 
-         std::set_intersection( _producers.begin(), _producers.end(),
-                                active_producers.begin(), active_producers.end(),
-                                boost::make_function_output_iterator( [&]( const chain::account_name& producer )
-         {
-            if( producer != bsp->header.producer ) {
-               auto itr = std::find_if( active_producer_to_signing_key.begin(), active_producer_to_signing_key.end(),
-                                        [&](const producer_key& k){ return k.producer_name == producer; } );
-               if( itr != active_producer_to_signing_key.end() ) {
-                  auto private_key_itr = _signature_providers.find( itr->block_signing_key );
-                  if( private_key_itr != _signature_providers.end() ) {
-                     auto d = bsp->sig_digest();
-                     auto sig = private_key_itr->second( d );
-                     _last_signed_block_time = bsp->header.timestamp;
-                     _last_signed_block_num  = bsp->block_num;
-
-   //                  ilog( "${n} confirmed", ("n",name(producer)) );
-                     _self->confirmed_block( { bsp->id, d, producer, sig } );
-                  }
-               }
-            }
-         } ) );
-
          // simplify handling of watermark in on_block
```

Comment:
It seems like we need `_producer_watermarks` to map to both the watermark block number and the watermark block time. We would update the watermark time both here and in `produce_block` (unless it would cause it to decrease).

Reply:
For the context of this PR I'm going to ignore this feedback as it should be placed into an issue and resolved in the context of working on watermark safety.